RPM 4.17.1 bugfix update

dmnks opened the pull request:
Here's a crafted list of cherry picks (and a few backports) for the first maintenance update of the 4.17 line. Note that we're only focusing on regression, security and other important user-visible fixes here. Some documentation updates are also included.

The selection is based on the log between the last cherry pick on the `rpm-4.17.x` branch (commit 2a03b8fa12319d4650ee3cd1c48d1cd3ab8afe56) and master (as of today).

Please voice your opinion about the selection of commits and suggest anything that we might have missed or should not include.

DemiMarie commented:
Can b447ad31fb4567c524938edd65112a6476113a2e and 0e9f6fdc63f09ebd00516e6f2f46a7297b743bcd be backported as well?

dmnks commented:
Yeah, those two seem reasonable, I'll include them, thanks.

pmatilai commented:
0e9f6fdc6 was part of the initial draft I saw, but seems to have gotten dropped here. b447ad31f is a bit trickier as we generally can't drop stuff in maintenance releases, undocumented or not.

dmnks commented:
> [0e9f6fd](https://github.com/rpm-software-management/rpm/commit/0e9f6fdc63f09ebd00516e6f2f46a7297b743bcd) was part of the initial draft I saw, but seems to have gotten dropped here.

Oh, indeed. I dropped it from the draft list in an attempt to trim down the number of picks. No big deal to bring it back.

> [b447ad3](https://github.com/rpm-software-management/rpm/commit/b447ad31fb4567c524938edd65112a6476113a2e) is trickier as we generally can't drop stuff in maintenance releases, undocumented or not.

Makes sense. Let's keep this one out then, I guess.

pmatilai commented:
Here's one more to include if not too late: https://github.com/rpm-software-management/rpm/pull/2100 This isn't of course enabled in any default config of rpm but still its ugly to have rpmbuild depend on network access.

dmnks commented:
Fixed the gcc deprecation warnings (by cherry-picking commit 6eec1f8a005db6ff9a1cce57afa5dba4253a06b2 as is, not just the header patch) and added a release preparation commit on top of the branch.

Rationale behind the `rpm_version_info` bump: we're adding new interfaces (commits 7fd7a60ea29aab018359037058de95b116455db1 and daeddb01de50c53b98eae75e11234132ae63538a) so incrementing _current_, resetting _revision_ to 0 and incrementing _age_. No interfaces are being removed.

All systems go. :rocket:

pmatilai reviewed (approved):
Ack.

I'll leave the honor of hitting the merge button on your first release for you :partying_face:

dmnks commented:
Thanks!
